### PR TITLE
www-misc/buku: Add version 4.3

### PR DESCRIPTION
**Pull request by ajakk**
Bug: https://bugs.gentoo.org/714590
Signed-off-by: John Helmert III <jchelmertt3@gmail.com>
Package-Manager: Portage-2.3.95, Repoman-2.3.21

### CODE REVIEW

**Comment by juippis** [comment]
@stkw0 what do you think? Would it make sense to remove baku-server to its own package for example?

@ajakk if there's no reply from @stkw0 in say 5-7 days, please ping us here again and we'll start making decisions.

File: www-misc/buku/buku-4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		dev-python/pytest[${PYTHON_USEDEP}]
+	)
+"
+
+python_prepare() {
+	distutils-r1_python_prepare_all
+
+	# Remove support for bukuserver - complex depgraph which isn't all
+	# sufficiently packaged in Gentoo
+	sed -ie '/console_scripts/s/,.*/]/' setup.py
```
Comment (juippis) [inline www-misc/buku/buku-4.3.ebuild]: Needs to be appended with `|| die`.

**Comment by stkw0** [comment]
There is already another PR for buku which have a USE flag to support 
bukuserver and it worked well (at least with the previous version).

Currently I don't have a stable internet connection and due covid19 it 
may take a while until my internet provider can fix it, but I would 
appreciate to be able to provide both buku and bukuserver (and most 
packages do it in this way, see for example gdb).

**Comment by ajakk** [comment]
I was actually working on hiding bukuserver behind a useflag simply because it would bring in [many more dependencies](https://github.com/jarun/buku/blob/v4.3/setup.py#L35), but a number of those dependencies are limited to Python 3.6 or 3.7 (and flask-admin is lastrited), so I came to the conclusion that it would be an unecessary headache to try to get those deps up to Python 3.8 just to have bukuserver.

I'd be happy to help work on getting the dependencies up to what they need to be, but I figured having bukuserver wasn't worth blocking the update completely due to its dependencies.

**Comment by stkw0** [comment]
I agree with you, if having bukuserver support requires much work it's 
better to remove it for now (and add it latter whenever possible).

Regards :)

File: www-misc/buku/buku-4.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	newbashcomp auto-completion/bash/buku-completion.bash "${PN}"
+
+	doman buku.1
+}
+
+python_test() {
+	# Explicitly enumerate tests to avoid tests that deal with bukuserver
+	tests=$(echo "tests/test_"{BukuCrypt,bukuDb,buku,ExtendedArgumentParser,import_firefox_json}".py")
+
+	py.test -v ${tests} || die
```
Comment (juippis) [inline www-misc/buku/buku-4.3.ebuild]: Is the variable needed here? I think you can just specify these files after `py.test`.

```
========================================== ERRORS ==========================================
____________________ ERROR at setup of test_load_firefox_database[True] ____________________

self = <urllib.request.HTTPSHandler object at 0x7f8442cde2e8>
http_class = <class 'http.client.HTTPSConnection'>
req = <urllib.request.Request object at 0x7f84422f2898>
http_conn_args = {'check_hostname': None, 'context': None}, host = 'github.com'
h = <http.client.HTTPSConnection object at 0x7f8442cde358>
...
...
tests/test_bukuDb.py:1190: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib64/python3.6/urllib/request.py:223: in urlopen
    return opener.open(url, data, timeout)
/usr/lib64/python3.6/urllib/request.py:526: in open
    response = self._open(req, data)
/usr/lib64/python3.6/urllib/request.py:544: in _open
    '_open', req)
/usr/lib64/python3.6/urllib/request.py:504: in _call_chain
    result = func(*args)
/usr/lib64/python3.6/urllib/request.py:1361: in https_open
    context=self._context, check_hostname=self._check_hostname)
...
...
tests/test_bukuDb.py:627: AssertionError
----------------------------------- Captured stderr call -----------------------------------
WARNING:urllib3.connectionpool:Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8442c930f0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com
WARNING:urllib3.connectionpool:Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8442c931d0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com
WARNING:urllib3.connectionpool:Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f8442ec3128>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',)': /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com
ERROR:root:HTTPSConnectionPool(host='tny.im', port=443): Max retries exceeded with url: /yourls-api.php?action=shorturl&format=simple&url=https%3A%2F%2Fwww.google.com (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f84431e6160>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',))
```
etc, could you get tests to run? `========== 17 failed, 344 passed, 2 skipped, 4 warnings, 2 error in 2.78 seconds ===========`

File: www-misc/buku/buku-4.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	>=dev-python/html5lib-1.0.1[${PYTHON_USEDEP}]
+	>=dev-python/beautifulsoup-4.6.0[${PYTHON_USEDEP}]
+	dev-python/cryptography[${PYTHON_USEDEP}]
+	dev-python/urllib3[${PYTHON_USEDEP}]
```
Comment (juippis) [inline www-misc/buku/buku-4.3.ebuild]: Please sort in alphabetical order.

File: www-misc/buku/buku-4.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	test? (
+		>=dev-python/click-7.0[${PYTHON_USEDEP}]
+		>=dev-python/py-1.5.0[${PYTHON_USEDEP}]
+		>=dev-python/pylint-1.7.2[${PYTHON_USEDEP}]
+		>=dev-python/pyyaml-4.2[${PYTHON_USEDEP}]
+		>=dev-python/vcrpy-4.0.2[${PYTHON_USEDEP}]
+		dev-python/attrs[${PYTHON_USEDEP}]
+		dev-python/flake8[${PYTHON_USEDEP}]
+		dev-python/hypothesis[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
```
Comment (juippis) [inline www-misc/buku/buku-4.3.ebuild]: Here too.

File: www-misc/buku/buku-4.3.ebuild (hunk 2 above)
Comment (ajakk) [reply www-misc/buku/buku-4.3.ebuild]: The tests work with FEATURES=-network-sandbox. It uses vcrpy but seemingly not for everything. I can ask upstream to fix it.

File: www-misc/buku/buku-4.3.ebuild (hunk 4 above)
Comment (ajakk) [reply www-misc/buku/buku-4.3.ebuild]: Perhaps I'm just dense, but aren't these already sorted? Should the versioned deps go after because of the ">"?

Comment (juippis) [reply www-misc/buku/buku-4.3.ebuild:38]: You should ignore the operators and sort by letters :)
```
		dev-python/attrs[${PYTHON_USEDEP}]
		>=dev-python/click-7.0[${PYTHON_USEDEP}]
		dev-python/flake8[${PYTHON_USEDEP}]
		dev-python/hypothesis[${PYTHON_USEDEP}]
		>=dev-python/py-1.5.0[${PYTHON_USEDEP}]
		>=dev-python/pylint-1.7.2[${PYTHON_USEDEP}]
		dev-python/pytest[${PYTHON_USEDEP}]
		>=dev-python/pyyaml-4.2[${PYTHON_USEDEP}]
		>=dev-python/vcrpy-4.0.2[${PYTHON_USEDEP}]
```

I know it looks better to you the way it is now, but it's better for US like this.

File: www-misc/buku/buku-4.3.ebuild (hunk 2 above)
Comment (juippis) [reply www-misc/buku/buku-4.3.ebuild:65]: I guess you'll need to restrict tests then for this release.

Comment (ajakk) [reply www-misc/buku/buku-4.3.ebuild:65]: I have an [open PR](https://github.com/jarun/buku/pull/443) and [open issue](https://github.com/jarun/buku/issues/442) upstream to fix it, but they've proven pretty hostile to network-sandbox.

File: www-misc/buku/buku-4.3.ebuild (hunk 4 above)
Comment (ajakk) [reply www-misc/buku/buku-4.3.ebuild:38]: I'm happy to sort however is wanted, I just didn't understand what that was. I'll fix it up and push in a bit.

**Pull request closed without merging**